integration-tests/src/test/java/org/openmrs/maven/plugins/SetupIT.java

```diff
@@ -527,6 +528,35 @@ public void setup_shouldInstallWithContentPackageWithNoNamespace() throws Except
         assertFilePresent(serverId, "configuration", "encountertypes", "encountertypes.csv");
     }
 
+    @Test
+    public void testSetupWithMissingContentDependencies() throws Exception {
+        setupContentPackage("testpackage1");
+
+        includeDistroPropertiesFile("openmrs-distro-content-package-missing-dependencies.properties");
+        addTaskParam("distro", testDirectory.toString() + File.separator + "openmrs-distro.properties");
+        addMockDbSettings();
+
+        String serverId = UUID.randomUUID().toString();
+        addAnswer(serverId);
+        addAnswer("1044");
+        addAnswer("8080");
+
+        Exception exception = null;
+        try {
+            executeTask("setup");
+        }
+        catch (Exception e) {
+            exception = e;
+        }
+        assertNotNull(exception);
+        assertLogContains("requires module org.openmrs.module:fhir2-omod version >=1.8.0");
+        assertLogContains("requires module org.openmrs.module:webservices.rest-omod version >=2.42.0");
+        assertLogContains("requires module org.openmrs.module:openconceptlab-omod version >=2.3.0");
+        assertLogContains("requires module org.openmrs.module:initializer-omod version >=2.5.2");
+        assertLogContains("requires module org.openmrs.module:o3forms-omod version >=2.2.0");
+        assertLogContains("requires esm @openmrs/esm-patient-chart-app version >=8.1.0");
+    }
+
```

Comment:
I have added integration tests for the 3 impacted goals with respect to content package validation.  In each case, the new default behavior is to fail, and provide a clear message as to why the build failed and what is needed to fix it.